### PR TITLE
Adding example on Physics-Informed Neural Networks for solving Eikonal equation

### PR DESCRIPTION
**Pull request by sgrubas**
Hello,

According to [my issue](https://github.com/keras-team/keras-io/issues/1287) (closed), you might be interested in a new example on physics-informed neural networks (PINNs).

I created a new folder `pinn` in the repo since it did not have any related to physics-informed neural networks (pinn)
This `.py` file contains the example with the neural eikonal solver.

Serafim

### CODE REVIEW

File: examples/pinn/eikonal.py [hunk 1]
```diff
@@ -0,0 +1,459 @@
...
+massive computations. The latter will be shown in the end of the example.
+"""
+
+"""
+## Imports
+"""
+
+import tensorflow as tf
+from tensorflow.keras.layers.experimental.preprocessing import Rescaling
+from tensorflow.keras.layers import Input, Concatenate, Dense
```
Comment (fchollet) [inline examples/pinn/eikonal.py:70]: Just `from tensorflow.keras import layers`, then access elements of the module, e.g. `layers.Dense`

File: examples/pinn/eikonal.py [hunk 2]
```diff
@@ -0,0 +1,459 @@
...
+from tensorflow.keras.layers import Multiply, Subtract, Lambda
+from tensorflow.keras.models import Model
```
Comment (fchollet) [inline examples/pinn/eikonal.py:72]: Access it from `keras`

File: examples/pinn/eikonal.py [hunk 3]
```diff
@@ -0,0 +1,459 @@
...
+used for any geometry and distribution of collocation points where we seek for solution.
+Another important advantage, it can provide dramatic compression of the solution in
+massive computations. The latter will be shown in the end of the example.
+"""
+
+"""
+## Imports
+"""
+
+import tensorflow as tf
```
Comment (fchollet) [inline examples/pinn/eikonal.py:68]: Also import `keras`

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: examples/pinn/eikonal.py [hunk 4]
```diff
@@ -0,0 +1,459 @@
+"""
+Title: Neural Eikonal Solver
```
[the remark this answers is not in the document]
Comment (sgrubas) [reply examples/pinn/eikonal.py:2]: Thanks for your comments and reply!
I was also thinking about "quick keras recipes" but alternatively since eikonal is widely used in computer vision if I adapt the example to a computer vision context (e.g. to Shape-from-Shading problem), do you think that "computer vision" could be a good fit?

[the remark this answers is not in the document]
Comment (LukeWood) [reply examples/pinn/eikonal.py:2]: I say we put it in quick keras recipes and focus on the concept of PINNs!
I think we should emphasize that in PINNs you simply add an L2 loss to enforce the implicit physics constraints of the system.

**Comment by sgrubas** [comment]
Hello, 
I revised the pull request in accordance with your comments, please review
Serafim

**Comment by divyashreepathihalli** [comment]
@sgrubas  sorry this PR has slipped through the cracks. Since Keras has changed from tf backend to multi-backend. If you are open to converting this to a Keras 3 example and submitting it please go ahead. Else, let us know if I can close this PR.

**Comment by sgrubas** [comment]
> @sgrubas  sorry this PR has slipped through the cracks. Since Keras has changed from tf backend to multi-backend. If you are open to converting this to a Keras 3 example and submitting it please go ahead. Else, let us know if I can close this PR.

Hello, thanks for letting me know.
I'd like to update this example to be a Keras 3 one. Should it work with all backends or can I keep it as an example for Tensorflow backend only?
